src/zenml/integrations/kubernetes/orchestrators/kubernetes_orchestrator.py

```diff
@@ -0,0 +1,384 @@
+            kubernetes_namespace=self.kubernetes_namespace,
+            pb2_pipeline=pb2_pipeline,
+            sorted_steps=sorted_steps,
+            pipeline_dag=pipeline_dag,
+        )
+
+        # Authorize orchestrator pod to run k8s commands inside the cluster.
+        service_account_name = "zenml-service-account"
+        core_api = kube_utils.make_core_v1_api()
+        kube_utils.create_edit_service_account(
+            core_api=core_api,
+            service_account_name=service_account_name,
+            namespace=self.kubernetes_namespace,
+        )
+
+        # Schedule as CRON job if CRON schedule is given.
+        if runtime_configuration.schedule:
+            if not runtime_configuration.schedule.cron_expression:
+                raise RuntimeError(
+                    "The kubernetes orchestrator only supports scheduling via "
+                    "CRON jobs, but the run was configured with a manual "
+                    "schedule. Use `Schedule(cron_expression=...)` instead."
+                )
+            cron_expression = runtime_configuration.schedule.cron_expression
+            cron_job_manifest = build_cron_job_manifest(
+                cron_expression=cron_expression,
+                run_name=run_name,
+                pod_name=pod_name,
+                pipeline_name=pipeline_name,
+                image_name=image_name,
+                command=command,
+                args=args,
+                service_account_name=service_account_name,
+            )
+            batch_api = k8s_client.BatchV1beta1Api()
+            batch_api.create_namespaced_cron_job(
+                body=cron_job_manifest, namespace=self.kubernetes_namespace
+            )
+            logger.info(
+                f"Scheduling kubernetes run `{pod_name}` with CRON expression "
+                f'`"{cron_expression}"`.'
+            )
+
+        # Create and run the orchestrator pod.
+        pod_manifest = build_pod_manifest(
+            run_name=run_name,
+            pod_name=pod_name,
+            pipeline_name=pipeline_name,
+            image_name=image_name,
+            command=command,
+            args=args,
+            service_account_name=service_account_name,
+        )
+        core_api.create_namespaced_pod(
+            namespace=self.kubernetes_namespace,
+            body=pod_manifest,
+        )
+
+        # Wait for the orchestrator pod to finish and stream logs.
+        if self.synchronous:
+            logger.info("Waiting for Kubernetes orchestrator pod...")
+            kube_utils.wait_pod(
+                core_api,
+                pod_name,
+                namespace=self.kubernetes_namespace,
+                exit_condition_lambda=kube_utils.pod_is_done,
+                stream_logs=True,
+            )
+        else:
+            logger.info(
+                f"Orchestration started asynchronously in pod "
+                f"`{self.kubernetes_namespace}:{pod_name}`"
```

Comment:
Maybe we can include a `kubectl logs ...` command in this log message that gives them the option to manually check the logs of the orchestrator pod?

Reply:
Agree, makes sense. I'll add that.